wa/workloads/speedometer/__init__.py

```diff
@@ -12,57 +12,259 @@
+        )
+        self.target.execute(browser_launch_cmd)
+        time.sleep(1)
+        self.target.execute("am force-stop {}".format(self.chrome_package))
+        time.sleep(1)
+
+        # Pull the preferences file from the device, modify it, and push it
+        # back.  This is done to bypass the 'first launch' screen of the
+        # browser we see after the cache is cleared.
+        self.preferences_xml = "{}_preferences.xml".format(self.chrome_package)
+
+        self.target.pull(
+            "/data/data/{}/shared_prefs/{}".format(
+                self.chrome_package, self.preferences_xml
+            ),
+            context.output_directory,
+            as_root=True,
+        )
+
+        with open(
+            os.path.join(context.output_directory, self.preferences_xml)
+        ) as read_fh:
+            lines = read_fh.readlines()
+
+            # Add additional elements for the preferences XML to the
+            # _second-last_ line
+            for line in [
+                '<boolean name="first_run_flow" value="true" />\n',
+                '<boolean name="first_run_tos_accepted" value="true" />\n',
+                '<boolean name="first_run_signin_complete" value="true" />\n',
+            ]:
+                lines.insert(len(lines) - 1, line)
+
+            with open(
+                os.path.join(context.output_directory, self.preferences_xml + ".new"),
+                "w",
+            ) as write_fh:
+                for line in lines:
+                    write_fh.write(line)
+
+        self.target.push(
+            os.path.join(context.output_directory, self.preferences_xml + ".new"),
+            "/data/data/{}/shared_prefs/{}".format(
+                self.chrome_package, self.preferences_xml
+            ),
+            as_root=True,
+        )
+
+    def run(self, context):
+        super(Speedometer, self).run(context)
+
+        browser_launch_cmd = "am start -a android.intent.action.VIEW -d {} {}".format(
+            self.speedometer_url, self.chrome_package
+        )
+        self.target.execute(browser_launch_cmd)
+
+        # Wait 60 seconds before we start polling the system to check which
+        # processes are busy.
+        time.sleep(60)
+
+        # We can only consider the benchmark complete and proceed if we see 5
+        # times _in a row_ that a process containing the string
+        # 'sandboxed_process' is not the hottest process in the system. We will
+        # sleep for 2 seconds between every check.
+        top_cmd = "{} top -b -n1".format(self.target.busybox)
+
+        def _top_output_for_busiest_process():
+            # busybox's top always outputs the Mem/CPU/Load/Header lines first,
+            # so line 5 is the hottest process.
+            return self.target.execute(top_cmd).split("\n")[4]
+
+        countdown = 5
+        while countdown > 0:
+            while "sandboxed_process" in _top_output_for_busiest_process():
```

Comment:
Testing this in chrome on my nexus 5 the process line is:
` 8845  1438 99021    S <  1651m 90.9   1 17.1 {dboxed_process0} com.android.chrome:sandboxed`
Do you have access to more devices to compare what the process appears as on other devices?